File: ios/Sources/ACFluentUI/FluentTypography.swift
import SwiftUI

public struct FluentTypography: Equatable {
    // Caption sizes
    public var caption2: FluentFont
    public var caption1: FluentFont
    
    // Body sizes
    public var body1: FluentFont
    public var body2: FluentFont
    
    // Subtitle sizes
    public var subtitle2: FluentFont
    public var subtitle1: FluentFont
    
    // Title sizes
    public var title3: FluentFont
    public var title2: FluentFont
    public var title1: FluentFont
    
    // Display sizes
    public var largeTitle: FluentFont
    public var display: FluentFont
    
    public init(
        caption2: FluentFont = FluentFont(size: 10, weight: .regular, lineHeight: 12),
        caption1: FluentFont = FluentFont(size: 12, weight: .regular, lineHeight: 16),
        body1: FluentFont = FluentFont(size: 14, weight: .regular, lineHeight: 20),
        body2: FluentFont = FluentFont(size: 14, weight: .semibold, lineHeight: 20),
        subtitle2: FluentFont = FluentFont(size: 16, weight: .regular, lineHeight: 22),
        subtitle1: FluentFont = FluentFont(size: 16, weight: .semibold, lineHeight: 22),
        title3: FluentFont = FluentFont(size: 20, weight: .semibold, lineHeight: 26),
        title2: FluentFont = FluentFont(size: 24, weight: .semibold, lineHeight: 32),
        title1: FluentFont = FluentFont(size: 28, weight: .semibold, lineHeight: 36),
        largeTitle: FluentFont = FluentFont(size: 32, weight: .semibold, lineHeight: 40),
        display: FluentFont = FluentFont(size: 40, weight: .semibold, lineHeight: 52)
    ) {
        self.caption2 = caption2
        self.caption1 = caption1
        self.body1 = body1
        self.body2 = body2
        self.subtitle2 = subtitle2
        self.subtitle1 = subtitle1
        self.title3 = title3
        self.title2 = title2
        self.title1 = title1
        self.largeTitle = largeTitle
        self.display = display
    }
}

public struct FluentFont: Equatable {
    public var size: CGFloat
    public var weight: Font.Weight
    public var lineHeight: CGFloat
    
    public init(size: CGFloat, weight: Font.Weight, lineHeight: CGFloat) {
        self.size = size
        self.weight = weight
        self.lineHeight = lineHeight
    }
    
    public var font: Font {
        .system(size: size, weight: weight)
    }
}

File: android/ac-fluent-ui/src/main/kotlin/com/microsoft/adaptivecards/fluentui/FluentTypography.kt
package com.microsoft.adaptivecards.fluentui

import androidx.compose.runtime.Immutable
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.sp

@Immutable
data class FluentTypography(
    // Caption sizes
    val caption2: FluentFont = FluentFont(size = 10.sp, weight = FontWeight.Normal, lineHeight = 12.sp),
    val caption1: FluentFont = FluentFont(size = 12.sp, weight = FontWeight.Normal, lineHeight = 16.sp),
    
    // Body sizes
    val body1: FluentFont = FluentFont(size = 14.sp, weight = FontWeight.Normal, lineHeight = 20.sp),
    val body2: FluentFont = FluentFont(size = 14.sp, weight = FontWeight.SemiBold, lineHeight = 20.sp),
    
    // Subtitle sizes
    val subtitle2: FluentFont = FluentFont(size = 16.sp, weight = FontWeight.Normal, lineHeight = 22.sp),
    val subtitle1: FluentFont = FluentFont(size = 16.sp, weight = FontWeight.SemiBold, lineHeight = 22.sp),
    
    // Title sizes
    val title3: FluentFont = FluentFont(size = 20.sp, weight = FontWeight.SemiBold, lineHeight = 26.sp),
    val title2: FluentFont = FluentFont(size = 24.sp, weight = FontWeight.SemiBold, lineHeight = 32.sp),
    val title1: FluentFont = FluentFont(size = 28.sp, weight = FontWeight.SemiBold, lineHeight = 36.sp),
    
    // Display sizes
    val largeTitle: FluentFont = FluentFont(size = 32.sp, weight = FontWeight.SemiBold, lineHeight = 40.sp),
    val display: FluentFont = FluentFont(size = 40.sp, weight = FontWeight.SemiBold, lineHeight = 52.sp)
)

@Immutable
data class FluentFont(
    val size: TextUnit,
    val weight: FontWeight,
    val lineHeight: TextUnit
)
